src/stdlib_string_type.fypp

```diff
@@ -721,6 +732,45 @@ contains
 
     end function verify_char_string
 
+    !> Moves the allocated character scalar from 'from' to 'to'
+    !> No output
+    subroutine move_string_string(from, to)
+        type(string_type), intent(inout) :: from
+        type(string_type), intent(out) :: to
+
+        call move_alloc(from%raw, to%raw)
```

Comment:
`move_alloc` requires its arguments to be `allocatable`. 
The `move` function arguments here are `type(string_type)`, which is not `allocatable`. 
If we overload `move_alloc`, it may cause users to misunderstand that `move_alloc` is different from the Fortran standard at `allocatable` attribute . 
From this perspective, I think the name can be `move` or `move_alloc_`, better not `move_alloc`?

Reply:
I found `move` to be intuitive as well. As if it cut and pastes (not exactly) a string from one location to other. Similar to moving a file from one folder to other.